wandb/sdk/internal/file_stream.py

```diff
@@ -249,43 +297,66 @@ class FileStreamApi:
     TODO: Differentiate between binary/text encoding.
     """
 
-    Finish = collections.namedtuple("Finish", ("exitcode"))
-    Preempting = collections.namedtuple("Preempting", ())
-    PushSuccess = collections.namedtuple("PushSuccess", ("artifact_id", "save_name"))
+    class Finish(NamedTuple):
+        exitcode: int
+
+    class Preempting(NamedTuple):
+        pass
+
+    class PushSuccess(NamedTuple):
+        artifact_id: str
+        save_name: str
 
     HTTP_TIMEOUT = env.get_http_timeout(10)
     MAX_ITEMS_PER_PUSH = 10000
 
-    def __init__(self, api, run_id, start_time, settings=None):
-        if settings is None:
-            settings = dict()
+    def __init__(
+        self,
+        api: "internal_api.Api",
+        run_id: str,
+        start_time: float,
+        settings: Optional[
+            Union[
+                "wandb.sdk.wandb_settings.Settings",
+                "settings_static.SettingsStatic",
```

Comment:
Is this right? isnt this settings just a dict? is this more aspirational than a reflection of the current code?

Reply:
Aspirational :) Made it into an `Optional[dict]`